staging/src/k8s.io/cli-runtime/pkg/genericclioptions/command_headers.go

```diff
@@ -77,3 +77,15 @@ func (c *CommandHeaderRoundTripper) ParseCommandHeaders(cmd *cobra.Command, args
 		c.Headers[kubectlCommandHeader] = strings.Join(cmdStrs, " ")
 	}
 }
+
+// CancelRequest is propagated to the Delegate RoundTripper within
+// if the wrapped RoundTripper implements this function.
+func (c *CommandHeaderRoundTripper) CancelRequest(req *http.Request) {
+	type canceler interface {
+		CancelRequest(*http.Request)
+	}
+	// If possible, call "CancelRequest" on the wrapped Delegate RoundTripper.
+	if cr, ok := c.Delegate.(canceler); ok {
```

Comment:
Is there a way to catch this at compile time or no because of how the delegators work?

Reply:
Unfortunately not. The `CancelRequest` function is not within the `RoundTripper` interface. It is a Kubernetes convention. Almost all Kubernetes RoundTrippers have the same ugly idiom.